EventsExpress.Core/Services/CategoryService.cs

```diff
@@ -6,41 +6,37 @@
 using EventsExpress.Core.DTOs;
 using EventsExpress.Core.Infrastructure;
 using EventsExpress.Core.IServices;
+using EventsExpress.Db.BaseService;
+using EventsExpress.Db.EF;
 using EventsExpress.Db.Entities;
-using EventsExpress.Db.IRepo;
 using Microsoft.EntityFrameworkCore;
 
 namespace EventsExpress.Core.Services
 {
-    public class CategoryService : ICategoryService
+    public class CategoryService : BaseService<Category>, ICategoryService
     {
-        private readonly IUnitOfWork _db;
-        private readonly IMapper _mapper;
-
-        public CategoryService(IUnitOfWork uow, IMapper mapper)
+        public CategoryService(AppDbContext context, IMapper mapper)
+            : base(context, mapper)
         {
-            _db = uow;
-            _mapper = mapper;
         }
 
-        public Category Get(Guid id) => _db.CategoryRepository.Get(id);
+        public Category GetById(Guid id)
+        {
+            return _context.Categories.Find(id);
+        }
 
         public IEnumerable<CategoryDTO> GetAllCategories()
         {
             var categories = _mapper.Map<List<CategoryDTO>>(
-                _db.CategoryRepository.Get()
-                .AsNoTracking()
-                .ToList());
+                _context.Categories.ToList());
 
             foreach (var cat in categories)
             {
-                cat.CountOfUser = _db.UserRepository.Get("Categories")
-                    .AsNoTracking()
+                cat.CountOfUser = _context.Users.Include("Categories")
```

Comment:
Are we using these properties anywhere? Seems like too much work to calculate them.